hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/dn/checksum/TestContainerCommandReconciliation.java

```diff
@@ -247,6 +251,35 @@ public void testGetChecksumInfoSuccess() throws Exception {
     }
   }
 
+  @Test
+  public void testDataChecksumReportedAtSCM() throws Exception {
+    long containerID = writeDataAndGetContainer(true);
+    // Check non-zero checksum after container close
+    Set<ContainerReplica> containerReplicas = cluster.getStorageContainerManager().getContainerManager()
+        .getContainerReplicas(ContainerID.valueOf(containerID));
+    for (ContainerReplica containerReplica: containerReplicas) {
+      assertNotEquals(0, containerReplica.getDataChecksum());
+    }
+    cluster.getStorageContainerLocationClient().reconcileContainer(containerID);
+    Thread.sleep(10000);
+
+    // Check non-zero checksum after container reconciliation
+    containerReplicas = cluster.getStorageContainerManager().getContainerManager()
+        .getContainerReplicas(ContainerID.valueOf(containerID));
+    for (ContainerReplica containerReplica: containerReplicas) {
+      assertNotEquals(0, containerReplica.getDataChecksum());
+    }
+
+    // Check non-zero checksum after datanode restart
+    // Restarting all the nodes take more time in mini ozone cluster, so restarting only one node
+    cluster.restartHddsDatanode(0, true);
+    containerReplicas = cluster.getStorageContainerManager().getContainerManager()
+        .getContainerReplicas(ContainerID.valueOf(containerID));
+    for (ContainerReplica containerReplica: containerReplicas) {
+      assertNotEquals(0, containerReplica.getDataChecksum());
```

Comment:
This may still be getting the old container info. There's no check that the DN has re-registered with SCM and sent the FCR between its restart and this check. The node will not have gone stale/dead in this time so [mini ozone's wait for node restart](https://github.com/apache/ozone/blob/86e483d9b24de9b95bc7093e4ac0e2071e77e913/hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneClusterImpl.java#L203) won't catch this condition

Reply:
The true flag ensures the Datanode becomes stale and removes it. During the startup it re-registers with the SCM. I have checked the log messages of these tests and that's what happens.

Reply:
SCM restart has been added to handle this case